Account/FirefoxAccount.swift

```diff
@@ -140,21 +149,26 @@ public class FirefoxAccount {
             configurationLabel = configurationLabel,
             email = dictionary["email"] as? String,
             uid = dictionary["uid"] as? String {
+                let fxaDeviceId = dictionary["fxaDeviceId"] as? String
+                let deviceRegistrationVersion = dictionary["deviceRegistrationVersion"] as? Int ?? 0
                 let stateCache = KeychainCache.fromBranch("account.state", withLabel: dictionary["stateKeyLabel"] as? String, withDefault: SeparatedState(), factory: stateFromJSON)
                 return FirefoxAccount(
                     configuration: configurationLabel.toConfiguration(),
                     email: email, uid: uid,
+                    fxaDeviceId: fxaDeviceId, deviceRegistrationVersion: deviceRegistrationVersion,
                     stateCache: stateCache)
         }
         return nil
     }
 
     public enum AccountError: MaybeErrorType {
         case NotMarried
+        case DeviceRegistrationFailed
```

Comment:
Note that Sync might introduce a fairly firm dependency on device registration — e.g., to list connected clients or send a push message — but that's fine, and doesn't imply using the FxA state machine to do the work.